src/toil/cwl/cwltoil.py

```diff
@@ -2081,14 +2082,21 @@ def __init__(
             )
 
         req = tool.evalResources(self.builder, runtime_context)
-
+        
+        accelerators: List[AcceleratorRequirement] = []
+        if req.get('cudaDeviceCount', 0) > 0:
+            # There's a CUDARequirement
+            # TODO: How is cwltool deciding what value to use between min and max?
+            accelerators.append({'kind': 'gpu', 'api': 'cuda', 'count': cast(int, req['cudaDeviceCount'])})
```

Comment:
> TODO: How is cwltool deciding what value to use between min and max?

Same as other resources

From https://www.commonwl.org/v1.0/CommandLineTool.html#ResourceRequirement
> If "min" is specified but "max" is not, then "max" == "min" If "max" is specified by "min" is not, then "min" == "max".

Implemented as https://github.com/common-workflow-language/cwltool/blob/fb816b477fe9f6f93e8ca8ce4b651351235549cd/cwltool/process.py#L1016

Reply:
I think my question @mr-c is: the workflow can specify both a min and a max, but here in `req['cudaDeviceCount']` I have a single number. If the workflow specifies both a min and a max and they are different numbers, which one will I see in `req['cudaDeviceCount']`, and how would I be able to get at the other one?

Reply:
Okay, then the default resource allocator has already smooshed the min/max into a single number at this point.

To have a custom allocator, set `select_resources` in the RuntimeContext to a function (prior to this point)

The function signature: https://github.com/common-workflow-language/cwltool/blob/fb816b477fe9f6f93e8ca8ce4b651351235549cd/cwltool/context.py#L133

Example: https://github.com/common-workflow-language/cwltool/blob/fb816b477fe9f6f93e8ca8ce4b651351235549cd/cwltool/executors.py#L286